### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by janisozaur**

### CODE REVIEW

File: src/openrct2/Context.cpp [hunk 1]
```diff
@@ -395,7 +395,7 @@ namespace OpenRCT2
 
         bool LoadParkFromStream(IStream * stream, const std::string &path, bool loadTitleScreenFirstOnFail) final override
         {
-            ClassifiedFileInfo info;
+            ClassifiedFileInfo info{};
```
Comment (IntelOrca) [inline src/openrct2/Context.cpp]: It would be better to give it default values in the struct definition.

File: src/openrct2/interface/Theme.cpp [hunk 2]
```diff
@@ -296,7 +296,7 @@ UIThemeWindowEntry UIThemeWindowEntry::FromJson(const WindowThemeDesc * wtDesc,
     uint8 numColours = (uint8)json_array_size(jsonColours);
     numColours = Math::Min(numColours, wtDesc->NumColours);
 
-    UIThemeWindowEntry result;
+    UIThemeWindowEntry result{};
```
Comment (IntelOrca) [inline src/openrct2/interface/Theme.cpp]: See above

File: src/openrct2/object/ObjectRepository.cpp [hunk 3]
```diff
@@ -709,7 +709,7 @@ extern "C"
                     Object * object = objectRepository->LoadObject(ori);
                     if (object != nullptr)
                     {
-                        StexObject * stexObject = static_cast<StexObject*>(object);
+                        StexObject * stexObject = dynamic_cast<StexObject*>(object);
```
Comment (IntelOrca) [inline src/openrct2/object/ObjectRepository.cpp]: Why change it to `dynamic_cast` if you aren't going to null check?

File: src/openrct2/ParkImporter.cpp [hunk 4]
```diff
@@ -117,22 +117,14 @@ namespace ParkImporter
 
     bool ExtensionIsRCT1(const std::string &extension)
     {
-        if (String::Equals(extension, ".sc4", true) ||
```
Comment (Gymnasiast) [inline src/openrct2/ParkImporter.cpp]: I find the old version easier to read.

File: src/openrct2/actions/GameAction.cpp [hunk 5]
```diff
@@ -27,8 +27,7 @@
 #include "../world/park.h"
 
 GameActionResult::GameActionResult()
-{
-}
+= default;
```
Comment (Gymnasiast) [inline src/openrct2/actions/GameAction.cpp]: This looks odd. And why on a new line?

File: src/openrct2/core/FileScanner.cpp [hunk 6]
```diff
@@ -67,7 +67,7 @@ class FileScannerBase : public IFileScanner
     {
         std::string                 Path;
         std::vector<DirectoryChild> Listing;
-        sint32                      Index;
+        sint32                      Index{};
```
Comment (Gymnasiast) [inline src/openrct2/core/FileScanner.cpp]: Can you put a space between `Index` and `{}`? (This applies to all occurrences, but I'll say this once.)

**Comment by ZehMatt** [comment]
I don't think everything should be null initialized like that. Especially char/wchar arrays that only require the first index to be null, doing the rest is a waste of cpu cycles. Plus it looks kinda awkward

File: src/openrct2/ride/Vehicle.cpp [hunk 7]
```diff
@@ -7448,12 +7444,7 @@ static bool loc_6DB38B(rct_vehicle * vehicle, rct_tile_element * tileElement)
     sint32 trackType   = track_element_get_type(tileElement);
     sint32 vangleStart = TrackDefinitions[trackType].vangle_start;
 
-    if (vangleStart != _vehicleVAngleEndF64E36 || bankStart != _vehicleBankEndF64E37)
-    {
-        return false;
-    }
-
-    return true;
+    return !(vangleStart != _vehicleVAngleEndF64E36 || bankStart != _vehicleBankEndF64E37);
```
Comment (marcovmun) [inline src/openrct2/ride/Vehicle.cpp]: Why not making this better readable and simpeler? Or do I miss something here
`return vangleStart == _vehicleVAngleEndF64E36 && bankStart == _vehicleBankEndF64E37`

Comment (Gymnasiast) [reply src/openrct2/ride/Vehicle.cpp]: Looks like you're right. ```!(!a || !b) === a && b```.

Comment (AaronVanGeffen) [reply src/openrct2/ride/Vehicle.cpp]: For what it's worth, this is one of [De Morgan's laws](https://en.wikipedia.org/wiki/De_Morgan%27s_laws#Formal_notation).

File: src/openrct2/ParkImporter.cpp (hunk 4 above)
Comment (thachachec) [reply src/openrct2/ParkImporter.cpp]: # LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!

File: src/openrct2/object/ObjectRepository.cpp (hunk 3 above)
Comment (thachachec) [reply src/openrct2/object/ObjectRepository.cpp]: # LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!

File: src/openrct2/interface/Theme.cpp (hunk 2 above)
Comment (thachachec) [reply src/openrct2/interface/Theme.cpp]: # LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!

File: src/openrct2/Context.cpp (hunk 1 above)
Comment (thachachec) [reply src/openrct2/Context.cpp]: # LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!

**Comment by thachachec** [comment]
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!

**Comment by thachachec** [comment]
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!
# LOL PINEAPPLE SEX!

**Comment by tobiaskohlbau** [comment]
Ping, as it contains lot of code quality improvements. @janisozaur what's up with this PR?

**Comment by Gymnasiast** [comment]
@tobiaskohlbau I think the code also contains lots of odd changes, and changes that decrease readability.

**Comment by janisozaur** [comment]
@tobiaskohlbau I will update it soon, I was busy getting SSE PR merged.

**Comment by tobiaskohlbau** [comment]
@Gymnasiast I don't get your point. It very much makes use of Core Guidelines which are a good reference if adopted. For e.g. what you called odd like:
```
void Class::~Destructor()
= default;
```

it's much better as the compiler generates the destructor and not an self defined empty destructor. (AKA trivial destructible objects). For your question about the new line: If you put it in the same line a future change would change the whole line within VSC whereas with new line it's enough to remove the default and put the implementation in. In addition I would prefer the more verbose way of returning the results instead of surrounding them with verbose if/else statements.

File: src/openrct2/core/FileScanner.cpp (hunk 6 above)
Comment (Broxzier) [reply src/openrct2/core/FileScanner.cpp]: It doesn't really make sense to initialize a type with `{}` to me, I'd rather see `= 0` here.

**Comment by Gymnasiast** [comment]
I have rebased this PR. Most of the changes had already crept in via other PRs/commits.

File: src/openrct2/drawing/X8DrawingEngine.cpp [hunk 8]
```diff
@@ -471,9 +471,7 @@ X8DrawingContext::X8DrawingContext(X8DrawingEngine * engine)
 }
 
 X8DrawingContext::~X8DrawingContext()
-{
-
-}
+= default;
```
Comment (Broxzier) [inline src/openrct2/drawing/X8DrawingEngine.cpp]: This destructor can be removed entirely.

File: src/openrct2/network/NetworkConnection.h [hunk 9]
```diff
@@ -58,7 +58,7 @@ class NetworkConnection final
 
 private:
     std::list<std::unique_ptr<NetworkPacket>>   _outboundPackets;
-    uint32                                      _lastPacketTime;
+    uint32                                      _lastPacketTime{};
```
Comment (Broxzier) [inline src/openrct2/network/NetworkConnection.h]: Another instance of using `{}` for initializing a type.

File: src/openrct2/network/NetworkGroup.cpp [hunk 10]
```diff
@@ -26,8 +26,7 @@ NetworkGroup::NetworkGroup()
 }
 
 NetworkGroup::~NetworkGroup()
-{
-}
+= default;
```
Comment (Broxzier) [inline src/openrct2/network/NetworkGroup.cpp]: Can be removed too.

File: src/openrct2/network/TcpSocket.cpp [hunk 11]
```diff
@@ -91,8 +94,7 @@ class TcpSocket final : public ITcpSocket
 
 public:
     TcpSocket()
-    {
-    }
+    = default;
```
Comment (Broxzier) [inline src/openrct2/network/TcpSocket.cpp]: new line before `= default`

File: src/openrct2/ride/TrackDesignRepository.cpp [hunk 12]
```diff
@@ -149,7 +149,7 @@ class TrackDesignRepository final : public ITrackDesignRepository
         Guard::ArgumentNotNull(env);
     }
 
-    ~TrackDesignRepository() = default;
+    ~TrackDesignRepository() override = default;
```
Comment (Broxzier) [inline src/openrct2/ride/TrackDesignRepository.cpp]: Can be removed.

**Comment by Broxzier** [comment]
The commit's description lists what gets changed exactly, the title needs to be generic since there are differnt kinds of changes made. `Code improvements` or something similar would be better I guess.

**Comment by AaronVanGeffen** [comment]
Ah, yours does list its changes, indeed. But yeah, "Various code improvements" would be better. To me "fixes" suggests bugs were addressed.

**Pull request merged**